src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py

```diff
@@ -64,18 +62,17 @@ def _extract_test_name(self, cmd_args: Dict[str, Any]) -> str:
                     return name.upper() if name.lower() == "gpt" else name.capitalize()
         return ""
 
-    def _update_env_vars(self, env_vars: Dict[str, str], cmd_args: Dict[str, Any], num_nodes: int):
-        """
-        Update environment variables.
+    def _update_env_vars(self, tr: TestRun):
+        """Update environment variables."""
+        env_vars = tr.test.test_definition.extra_env_vars
+        cmd_args = tr.test.test_definition.cmd_args_dict
+        num_nodes = len(tr.nodes) if tr.nodes else tr.num_nodes
 
-        Args:
-            env_vars (Dict[str, str]): Environment variables.
-            cmd_args (Dict[str, str]): Command-line arguments.
-            num_nodes (int): Number of nodes to use.
-        """
         self._update_per_gpu_combine_threshold(env_vars, cmd_args, num_nodes)
         self._update_xla_flags(env_vars, cmd_args)
 
+        tr.test.test_definition.extra_env_vars = env_vars
```

Comment:
I don't like the idea of full override of TOML/user defined arguments. Can use a more granular `.update` or `.extra_env_args[<ENV VAR>] = <VALUE>`?

Reply:
Please find the updated code.